src/access/wac.ts

```diff
@@ -314,10 +337,26 @@ async function setActorAccess<T extends WithServerResourceInfo>(
     resourceWithAcl,
     resourceAcl
   );
-  const currentAccess = readAccess(resourceWithOldAcl, actor) as AclAccess;
-  const wacAccess = universalAccessToAcl(access, currentAccess);
 
-  const updatedResourceAcl = writeAccess(resourceAcl, actor, wacAccess);
+  let currentAccess: AclAccess;
+  let updatedResourceAcl: (AclDataset & WithChangeLog) | undefined = undefined;
+  let wacAccess: AclAccess | undefined = undefined;
+
+  if (actor !== undefined) {
+    currentAccess = readAccess(resourceWithOldAcl, actor) as AclAccess;
+    wacAccess = universalAccessToAcl(access, currentAccess);
+    // @ts-ignore
+    updatedResourceAcl = writeAccess(resourceAcl, actor, wacAccess);
+  } else {
+    // In this case, no actor means we're using the Agent Class read/write Access,
+    // but TS still expects the Agent/Group read/write, with a slightly different
+    // signature.
+    // @ts-ignore
```

Comment:
I'd also generally prefer to avoid `@ts-ignore` and use the appropriate type assertion instead, to make sure we're not suppressing unrelated errors. So I think you'd cast `readAccess` to `typeof getPublicAccessWac` here. Though again, special-casing for one or two calling functions I think is a code smell.